### PR TITLE
Adding types for `expired-storage` npm package

### PR DESCRIPTION
**Pull request by intolerance**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by plantain-00 — CHANGES_REQUESTED** [review]

File: types/expired-storage/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": false,
```
[the remark this answers is not in the document]
Comment (intolerance) [reply types/expired-storage/tsconfig.json]: Thanks for your review. I've set `strictNullChecks` to `true`.

File: types/expired-storage/tslint.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+    "extends": "dtslint/dt.json",
```
[the remark this answers is not in the document]
Comment (intolerance) [reply types/expired-storage/tslint.json]: tslint now purely uses `dt.json` with no exceptions.

**Review by plantain-00 — APPROVED** [review]

File: types/expired-storage/expired-storage-tests.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import ExpiredStorage = require('expired-storage');
+
+// const localStorage = new LocalStorage('./scratch', Infinity);
+// const expiredStorage = new ExpiredStorage(localStorage); // ExpiredStorage requires a LocalStorage provider
+const expiredStorage = new ExpiredStorage();
+
+const mjkModel = {
+  firstName: 'Maynard',
+  middleName: 'James',
+  familyName: 'Keenan',
```
Comment (DanielRosenwasser) [inline types/expired-storage/expired-storage-tests.ts]: Nice.

File: types/expired-storage/expired-storage-tests.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
+// import { LocalStorage } from 'node-localstorage'; // Removed since this pkg isn't getting installed.
+import ExpiredStorage = require('expired-storage');
+
+// const localStorage = new LocalStorage('./scratch', Infinity);
+// const expiredStorage = new ExpiredStorage(localStorage); // ExpiredStorage requires a LocalStorage provider
```
Comment (DanielRosenwasser) [inline types/expired-storage/expired-storage-tests.ts]: Can you just remove the comments?

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

**Comment by DanielRosenwasser** [comment]
Thanks!

**Pull request merged**